File: features/fixtures/mazerunner/cxx-scenarios/src/main/java/com/bugsnag/android/mazerunner/scenarios/CXXThrowSomethingOutsideReleaseStagesScenario.java
package com.bugsnag.android.mazerunner.scenarios;

import com.bugsnag.android.Configuration;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import java.util.Collections;

public class CXXThrowSomethingOutsideReleaseStagesScenario extends Scenario {

    static {
        System.loadLibrary("cxx-scenarios");
    }

    public native void crash(int num);

    /**
     * Sets custom enabled release stages.
     */
    public CXXThrowSomethingOutsideReleaseStagesScenario(@NonNull Configuration config,
                                                         @NonNull Context context,
                                                         @Nullable String eventMetadata) {
        super(config, context, eventMetadata);
        config.setAutoTrackSessions(false);
        config.setEnabledReleaseStages(Collections.singleton("fee-fi-fo-fum"));
    }

    @Override
    public void startScenario() {
        super.startScenario();
        String metadata = getEventMetadata();
        if (metadata != null && metadata.equals("non-crashy")) {
            return;
        }
        crash(23);
    }
}

File: features/fixtures/mazerunner/cxx-scenarios/src/main/java/com/bugsnag/android/mazerunner/scenarios/CXXThrowSomethingOutsideReleaseStagesScenario.kt
package com.bugsnag.android.mazerunner.scenarios

import android.content.Context
import com.bugsnag.android.Configuration
import com.bugsnag.android.mazerunner.getZeroEventsLogMessages

class CXXThrowSomethingOutsideReleaseStagesScenario(
    config: Configuration,
    context: Context,
    eventMetadata: String?
) : Scenario(config, context, eventMetadata) {

    /**
     * Sets custom enabled release stages.
     */
    init {
        config.enabledReleaseStages = setOf("fee-fi-fo-fum")
        System.loadLibrary("cxx-scenarios")
    }

    external fun crash(num: Int)

    override fun startScenario() {
        super.startScenario()
        crash(23)
    }

    override fun getInterceptedLogMessages(): List<String> {
        return getZeroEventsLogMessages(startBugsnagOnly)
    }
}
